### PR TITLE
Coding Question Practice

### PR DESCRIPTION
**Pull request by Hojarbu**

### CODE REVIEW

File: translations/README-uz.md [hunk 1]
```diff
@@ -1257,46 +1257,47 @@ Computer science da graflar bir talay masalalarni tasvirlashda ishlatiladi. Shun
 
 ---
 
-## Coding Question Practice
+## Coding bo'yicha savollar ustida ishlash
 
-Now that you know all the computer science topics above, it's time to practice answering coding problems.
+Yuqoridagi barcha "computer science" mavzularini bilb oldingiz, endi coding bo'yicha masalalarga javob berishni mashq qilish vaqti keldi.
```
Comment (vovopap) [inline translations/README-uz.md:1261]: bilib

File: translations/README-uz.md [hunk 2]
```diff
@@ -1257,46 +1257,47 @@ Computer science da graflar bir talay masalalarni tasvirlashda ishlatiladi. Shun
...
 
-**Coding question practice is not about memorizing answers to programming problems.**
+**Codingda masalalarini yechishga tayyorlanish bu javoblarni yodlab qolish emas **
```
Comment (vovopap) [inline translations/README-uz.md]: Coding masalalarini...

File: translations/README-uz.md [hunk 3]
```diff
@@ -1257,46 +1257,47 @@ Computer science da graflar bir talay masalalarni tasvirlashda ishlatiladi. Shun
...
-- Problem recognition, and where the right data structures and algorithms fit in
-- Gathering requirements for the problem
-- Talking your way through the problem like you will in the interview
-- Coding on a whiteboard or paper, not a computer
-- Coming up with time and space complexity for your solutions
-- Testing your solutions
+Masalalar yechishni mashq qilish nima uchun zarur:
+- Muammoni aniqlash va kerakli ma'lumotlar tuzilmalari va algoritmlari mos keladigan joyni topa olish
+- Masala uchun kerakli talablarni to'plash
+- Xuddi intervyudagi kabi o'zingizni usulingizni muammoni hal qilishda ko'rsatish
```
Comment (vovopap) [inline translations/README-uz.md]: Sal noaniq gap bo'libdi. Suggestion: Xuddi intervyudagi kabi muammoni yechish uchun qadam va qadam yurish

File: translations/README-uz.md [hunk 4]
```diff
@@ -1257,46 +1257,47 @@ Computer science da graflar bir talay masalalarni tasvirlashda ishlatiladi. Shun
...
+- Doskada yoki qog'ozda kod yozish, kompyuterda emas
+- Ish joyi xarajati va ish vaqti xarajatini inobatga olgan holda masalaga yechim topish
```
Comment (vovopap) [inline translations/README-uz.md]: Bu yerda time and space complexityni inobatga olgan holda yechish emas, yechimlarizga time and space complexity qancha bulishini topish deyapti. Suggestion: "Yechimlarizga ish joyi xarajati va ish vaqti xarajatini aniqlash"

File: translations/README-uz.md [hunk 5]
```diff
@@ -1257,46 +1257,47 @@ Computer science da graflar bir talay masalalarni tasvirlashda ishlatiladi. Shun
...
+Siz buni dasturlash uchun intervyu kitoblaridan ham topa olishingiz mumkin
+lekin men buni ular ichida juda foydali deb topdim:
 [Algorithm design canvas](http://www.hiredintech.com/algorithm-design/)
 
-No whiteboard at home? That makes sense. I'm a weirdo and have a big whiteboard. Instead of a whiteboard, pick up a
-large drawing pad from an art store. You can sit on the couch and practice. This is my "sofa whiteboard".
-I added the pen in the photo for scale. If you use a pen, you'll wish you could erase. Gets messy quick. I use a pencil
-and eraser.
+Uyingizda doska yo'qmi? Bu tabiy. Men o'zgacha insonman, menda kotta doskam bor. Doska o'rniga, rasm chizish uchun 
+mo'ljallangan planshet oling. SHunda divanda o'tirib, mashq qilishingiz mumkin. Bu mening "divanga mo'ljallangan doskam".
```
Comment (vovopap) [inline translations/README-uz.md]: planshet to'gri kelmasa kerak bu yerda, u katta oq varaqni nazarda tutyapti

Comment (vovopap) [reply translations/README-uz.md]: https://d3j2pkmjtin6ou.cloudfront.net/art_board_sm_2.jpg

File: translations/README-uz.md [hunk 6]
```diff
@@ -1257,46 +1257,47 @@ Computer science da graflar bir talay masalalarni tasvirlashda ishlatiladi. Shun
...
+Men rasmda qalamimni ham qo'shganman qulaylik uchun. Agar siz ruchkadan foydalansangiz, o'chirish istagi paydo bo'lishi mumkin. 
+Tez Chalkashib ketadi. Shuning uchun men qalam va o'chirg'ichdan foydalanaman.
 
-![my sofa whiteboard](https://d3j2pkmjtin6ou.cloudfront.net/art_board_sm_2.jpg)
+![mening_divan_uchun doskam](https://d3j2pkmjtin6ou.cloudfront.net/art_board_sm_2.jpg)
```
Comment (vovopap) [inline translations/README-uz.md]: birinchi marta kurishim bunday yozishni xxxx_xxx_xxx. Shunday narsa bormi uzbek tilida?

File: translations/README-uz.md [hunk 7]
```diff
@@ -1257,46 +1257,47 @@ Computer science da graflar bir talay masalalarni tasvirlashda ishlatiladi. Shun
...
-Supplemental:
+Qo'shimcha:
 
 - [Mathematics for Topcoders](https://www.topcoder.com/community/competitive-programming/tutorials/mathematics-for-topcoders/)
 - [Dynamic Programming – From Novice to Advanced](https://www.topcoder.com/community/competitive-programming/tutorials/dynamic-programming-from-novice-to-advanced/)
 - [MIT Interview Materials](https://web.archive.org/web/20160906124824/http://courses.csail.mit.edu/iap/interview/materials.php)
 - [Exercises for getting better at a given language](http://exercism.io/languages)
 
-**Read and Do Programming Problems (in this order):**
+**Dasturlash uchun muammolarni o'qing va bajaring (ketma ketlikda):**
```
Comment (vovopap) [inline translations/README-uz.md]: Programming problems = Dasturlash masalalari

**Review by vovopap — APPROVED** [review]

File: translations/README-uz.md (hunk 6 above)
Comment (Hojarbu) [reply translations/README-uz.md]: ))) link buzilib ketishidan qo'rqib shunaqa yozibman

**Pull request merged**